WordPress/openverse
# Ariakit vue exploration with port of Heading

## Description


This PR is an exploration of writing Ariakit's components in Vue. I'm mostly using Openverse as a convenient development environment for testing, without needing to scaffold all the Vue compiler things in the ariakit repo (which does not look like it will be simple).

I'm also exploring ways of implementing the composition model that Ariakit relies on, namely the `as` prop. Vue requires explicit declarations for the top-level named props of a component, and does not support conditional prop types, so it isn't possible to follow ariakit React's model of joining the `as` component's props with the ariakit component's props.

Another thing I'd like to explore further is whether certain components are better suited as directives, or could be supported as a directive _or_ a component. I tried to find a way to do that with heading, but because it relies on the injected heading context, and directives don't support inject/provide, it isn't possible (at least not without ugly hacks!).

## Testing Instructions

Check out the AHeading and AHeadingLevel component. Compare them to https://ariakit.org/components/heading. Try them out in Storybook (`ov j p frontend storybook`) and on the 404 page (`ov nuxt` and visit localhost:8443/does-not-exist).

## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
